### PR TITLE
Fix attribute completion following JSX exprs

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes bug #3695

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -3059,30 +3060,22 @@ namespace ts {
...
-                                // Cursor is inside a JSX element
-                                attrsType = typeChecker.getJsxElementAttributesType((<JsxElement>jsxNode).openingElement);
-                            }
-                            symbols = typeChecker.getPropertiesOfType(attrsType);
-                            isMemberCompletion = true;
-                            return true;
-                        }
-                        jsxNode = jsxNode.parent;
+                    if (attrsType) {
+                        symbols = typeChecker.getPropertiesOfType(attrsType);
```
Comment (mhegazy) [inline src/services/services.ts]: you probably should filter these based on the ones already defined. you do not want to keep suggesting the same property over and over. i am assuming it is expected to define the same property multiple time outside a spread.

Comment (mhegazy) [inline src/services/services.ts]: e.g. filterObjectMembersList and filterModuleExports

File: src/services/services.ts [hunk 2]
```diff
@@ -3059,30 +3060,22 @@ namespace ts {
...
-                else if (getAncestor(contextToken, SyntaxKind.JsxElement) || getAncestor(contextToken, SyntaxKind.JsxSelfClosingElement)) {
-                    // Go up until we hit either the element or expression
-                    let jsxNode = contextToken;
+                else if(jsxContainer) {
+                    let attrsType: Type;
+                    if (jsxContainer.kind === SyntaxKind.JsxSelfClosingElement) {
+                        // Cursor is inside a JSX self-closing element
+                        attrsType = typeChecker.getJsxElementAttributesType(<JsxSelfClosingElement>jsxContainer);
+                    }
+                    else if(jsxContainer.kind === SyntaxKind.JsxOpeningElement) {
```
Comment (mhegazy) [inline src/services/services.ts]: These look identical.  a single if statement with an || in the condition.

Comment (RyanCavanaugh) [inline src/services/services.ts]: :heavy_check_mark:

File: src/services/services.ts (hunk 1 above)
Comment (RyanCavanaugh) [inline src/services/services.ts]: :heavy_check_mark: added a verification to the test as well

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**